server/app/views/docs/SchemaView.java

```diff
@@ -0,0 +1,232 @@
+        .withClasses("flex", "flex-row", "mt-2", "ml-2");
+  }
+
+  private static ImmutableList<OptionTag> getProgramSlugOptions(
+      Form form, ImmutableSet<String> allProgramSlugs) {
+    return allProgramSlugs.stream()
+        .sorted()
+        .map(x -> option(x).withValue(x).withCondSelected(x.equalsIgnoreCase(form.programSlug)))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  private static ImmutableList<OptionTag> getStageOptions(Form form) {
+    return ImmutableList.of(new F.Tuple<>("Active", "active"), new F.Tuple<>("Draft", "draft"))
+        .stream()
+        .map(
+            x ->
+                option(x._1)
+                    .withValue(x._2)
+                    .withCondSelected(x._2.equalsIgnoreCase(form.stage.orElse(""))))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  private static ImmutableList<OptionTag> getOpenApiOptions(Form form) {
+    return Arrays.stream(OpenApiVersion.values())
+        .sorted(Comparator.comparing(OpenApiVersion::toString))
+        .map(
+            x ->
+                option(x.getVersionNumber())
+                    .withValue(x.toString())
+                    .withCondSelected(
+                        x.toString().equalsIgnoreCase(form.openApiVersion.orElse(""))))
+        .collect(ImmutableList.toImmutableList());
+  }
+
+  /** Build the bootstrap bits that will generate the swagger/openapi UI */
+  private DivTag buildOpenApiDisplay(String apiUrl, String cspNonce) {
+    String swaggerUiId = "swagger-ui";
+
+    return div(
+            style(rawHtml(".swagger-ui > .topbar { display: none; }"))
+                .withType("text/css")
+                .attr("nonce", cspNonce),
+            div().withId(swaggerUiId),
+            script(
+                    rawHtml(
+                        """
+document.addEventListener('DOMContentLoaded', () => {
+    SwaggerUIBundle({
+        url: '%s',
+        dom_id: '#%s',
+        presets: [
+            SwaggerUIBundle.presets.apis,
+            SwaggerUIStandalonePreset
+        ],
+        layout: 'StandaloneLayout',
+        responseInterceptor: (response) => {
+          if (response.status === 404) {
+              console.log('404');
+              const uiError = document.querySelector('.swagger-ui.swagger-container');
+              const customErrorMessage = 'A program with this status could not be found.';
+              uiError.innerHTML = `<div data-testid="ui-error" style="display: flex; justify-content: center; margin: 4rem; font-size: 1.6rem;">${customErrorMessage}</div>`;
+          }
+          return response;
+        }
+    });
+
+    document.getElementById('programSlug')?.addEventListener('change', () => document.getElementById('form').submit());
+    document.getElementById('stage')?.addEventListener('change', () => document.getElementById('form').submit());
+    document.getElementById('openApiVersion')?.addEventListener('change', () => document.getElementById('form').submit());
```

Comment:
any particular reason we want this here and not in the admin js bundle? cause we don't want it on every admin page?

Reply:
Yeah I figured it'd be better to keep it localized to this file for that reason. I can change it if there's a strong feeling to do so, but it'll also require fiddling with webpack to get it to work correctly.